metals/src/main/scala/scala/meta/internal/metals/Compilers.scala

```diff
@@ -409,6 +410,21 @@ class Compilers(
     }
   }.getOrElse(Future.successful(Nil.asJava))
 
+  def extractMethod(
+      params: TextDocumentIdentifier,
+      range: LspRange,
+      defnPos: LspRange,
+      token: CancelToken,
+  ): Future[ju.List[TextEdit]] = {
+    withPCAndAdjustLsp(params, range) { (pc, pos, adjust) =>
+      pc.extractMethod(CompilerRangeParams.fromPos(pos, token), range, defnPos)
```

Comment:
```suggestion
      pc.extractMethod(CompilerRangeParams.fromPos(pos, token), range, adjust.adjustPos(defnPos))
```
might be needed in ammonite files